StoreProvider.js

```diff
@@ -0,0 +1,78 @@
+import { createContext, useContext, useState } from "react";
+import * as Location from "expo-location";
+import { Alert } from "react-native";
+const WeatherContext = createContext();
+
+const StoreProvider = ({ children }) => {
+  const [searchText, setSearchText] = useState("");
+  const [location, setLocation] = useState({});
+  const [showModal, setShowModal] = useState(false);
+  const [weatherDetails, setWeatherDetails] = useState({});
+  const [isLoading, setIsLoading] = useState(true);
+  const [isAnimating, setIsAnimating] = useState(false);
+  const randomCityNamesArr = [
+    "New Mumbai",
+    "New York City",
+    "Pune",
+    "Paris",
+    "Sydney ",
+    "London",
+  ];
+  const getRandomCityName = () => {
+    const randomNum = Math.floor(Math.random() * 6);
+    return randomCityNamesArr[randomNum];
+  };
+
+  const getUserLocation = async () => {
+    try {
+      setIsLoading(true);
+      const { status } = await Location.requestForegroundPermissionsAsync();
+      if (status !== "granted") {
```

Comment:
if status is something else, should we handle that as well? 
if permission is not given, what status comes? 
it may go in the catch block and handle, but i would like it to be optimistically handled. 
catch is for hardware level errors, never like an intended path. 

and if someone doesnt give location they should be able to use the app